### PR TITLE
Catalog: add deduplication for logs join

### LINKED ISSUE #1637: Refresh Locations grow each poll causing a memory leak
## Expected Behavior
- Each location is updated only once every poll
## Current Behavior
- When there are entities in the catalog, the polling time refresh grows exponentially and polls every entry a lot of times according to the log.

## Possible Solution
- Only poll each location once

## Steps to Reproduce
- Start the frontend and the backend
- Load the mock data `yarn lerna run mock-data`
- Watch the log and see the exponential locations grow.

```
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:07.088Z"}
info: Visiting 40 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:07.090Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:17.595Z"}
info: Visiting 80 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:17.597Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:28.536Z"}
info: Visiting 160 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:28.538Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:40.340Z"}
info: Visiting 294 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:40.344Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:53.573Z"}
info: Visiting 554 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:22:53.581Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:23:09.399Z"}
info: Visiting 820 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:23:09.409Z"}
info: Beginning locations refresh {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:23:27.252Z"}
info: Visiting 846 locations {"type":"plugin","plugin":"catalog","service":"backstage","timestamp":"2020-07-15T09:23:27.266Z"}
```

### PR DESCRIPTION
**Pull request by shmidt-i**
Fixes #1637 

Result of latest changes in migrations, needed one more layer of deduplication for `locations_update_log_latest` join.

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Prettier run on changed files
- [x] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ exports.up = async function up(knex) {
     ) t2
     ON t1.location_id = t2.location_id
     AND t1.created_at = t2.MAXDATE
+    GROUP BY t1.location_id
```
Comment (freben) [inline plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js:36]: Just noting that those that have existing persistent databases won't be picking this up. Are we at the point where we can't retrofit migrations yet? Not sure

**Review by freben — APPROVED** [review]

File: plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js (hunk 1 above)
Comment (shmidt-i) [reply plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js:36]: This is basically override for the override that we merged couple of weeks ago, should be okay. But - fair point, maybe we need to settle on migrations workflow going forward

Comment (shmidt-i) [reply plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js:36]: On a second thought, we already have at least couple of people that I'm aware of that do use persistent dbs. Will think it over and bring it tomorrow during sync ;)

Comment (Rugvip) [reply plugins/catalog-backend/migrations/20200527114117_location_update_log_latest_view.js:36]: Didnt fully settle but last week we kinda arrived at that we want to only append mugrations for now, but then flatten once we do a beta release or something like it

**Pull request merged**